proto_proxy.c

```diff
@@ -4053,6 +4111,15 @@ static int mcplib_open_dist_jump_hash(lua_State *L) {
 
 /*** END jump consistent hash library ***/
 
+/*** START lua interface to logger ***/
+
+static int mcplib_log(lua_State *L) {
+    LIBEVENT_THREAD *t = lua_touserdata(L, lua_upvalueindex(MCP_THREAD_UPVALUE));
+    const char *msg = luaL_checkstring(L, -1);
+    LOGGER_LOG(t->l, LOG_PROXYUSER, LOGGER_PROXY_USER, NULL, msg);
```

Comment:
I'm just confirming that the timestamp here is the timestamp at the time of the log being put onto the buffer, and not the time that the log is consumed.
Im 99% sure the former is the case, but just confirming.

Reply:
Log line's timestamp is when the log is inserted into the buffer. The elapsed time for this log entry is between the initial request generation and the result being returned to the client.